internal/delivery/http/v1/auth.go

```diff
@@ -24,8 +24,8 @@ func (h *Handler) initAuthRoutes(api *gin.RouterGroup) {
 // @Accept  json
 // @Produce  json
 // @Param input body service.SignupUserInput true "New user signup details"
```

Comment:
Swagger definitions below are incorrect:
- success does not return an object
- there is no 404 response, but there should be a 400 response for validation failure
- I recommend using utils.ValidationError as return type for 400 response, as this is what h.handleServiceError() writes to output on a validation error

In addition to that: 
- I recommend changing the success code to 204 (http.StatusNoContent) and change the swagger and the method code accordingly.
- there is a new method h.parseRequestBody() (in helpers.go) which I recommend using here to reduce code duplication

Reply:
Addressed in the latest commit.